### PR TITLE
Remove Outdated react lifecycle components & updated the react version to 16.4.1

### PR DESCRIPTION
**Pull request by superhit0**
@patw0929

### CODE REVIEW

File: src/components/IntlTelInputApp.js [hunk 1]
```diff
@@ -152,41 +171,29 @@ class IntlTelInputApp extends Component {
...
-      typeof nextProps.customPlaceholder === 'function' &&
-      this.props.customPlaceholder !== nextProps.customPlaceholder
-    ) {
-      this.updatePlaceholder(nextProps);
-    }
-  }
-
-  componentWillUpdate(nextProps, nextState) {
+  shouldComponentUpdate(nextProps, nextState) {
     if (nextState.showDropdown) {
```
Comment (patw0929) [inline src/components/IntlTelInputApp.js]: Although it's work, but I'm not sure if it is an anti-pattern to put codes which is not related to component-update control. 🤔

Comment (superhit0) [reply src/components/IntlTelInputApp.js]: componentWillUpdate is gone. 

The only place left for component will update code to go is in `should component update`

Comment (patw0929) [reply src/components/IntlTelInputApp.js]: OK, got it. 👌

**Comment by superhit0** [comment]
Merging it !!

**Pull request merged**